pkg/collector/check/stats/stats.go

```diff
@@ -118,7 +119,7 @@ type Stats struct {
 	LastDelay                int64     // most recent check start time delay relative to the previous check run, in seconds
 	LastWarnings             []string  // warnings that occurred in the last run, if any
 	UpdateTimestamp          int64     // latest update to this instance, unix timestamp in seconds
-	m                        sync.Mutex
+	M                        sync.Mutex
```

Comment:
Nit Perhaps it would be cleaner to keep this unexported, and instead introduce a helper function `SetStateCancelling` (or similar) that hides the locking?